src/json-api/json-api-module.ts

```diff
@@ -394,7 +394,7 @@ export class JsonApiModule {
   }
 
   public toApiNotificationConfiguration(configuration: NotificationConfiguration): ApiNotificationConfiguration {
-    return configuration as ApiNotificationConfiguration;
+    return { ...configuration } as ApiNotificationConfiguration;
```

Comment:
That's actually an excellent point. It should and it hides. This is something that's been unclear to me, i.e. these explicit vs. implicit casts in TypeScript. If I'm not mistaken, casting with `as` doesn't work in every situation and I don't really know what are the situations when it doesn't work. Here's some discussion about these matters: https://github.com/Microsoft/TypeScript/issues/7481.